src/main/kotlin/app/revanced/patcher/apk/ResourceFile.kt

```diff
@@ -0,0 +1,86 @@
+package app.revanced.patcher.apk
+
+import com.reandroid.xml.XMLDocument
+import com.reandroid.xml.XMLException
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.Closeable
+import java.io.InputStream
+import java.io.OutputStream
+
+/**
+ * A resource file inside an [Apk].
+ */
+class ResourceFile private constructor(
```

Comment:
This class does not belong in ReVanced Patcher because it is strictly dealing with resource files from arsclib. Please create a separate module that wraps around arsclib and use it in ReVanced Patcher.